fastcrypto-tbls/src/polynomial.rs

```diff
@@ -132,10 +126,14 @@ impl<C: GroupElement> Poly<C> {
     }
 
     /// Checks if a given share is valid.
-    pub fn is_valid_share(&self, idx: ShareIndex, share: &C::ScalarType) -> bool {
+    pub fn is_valid_share(&self, idx: ShareIndex, share: &C::ScalarType) -> FastCryptoResult<()> {
```

Comment:
I think the method should be renamed to `verify_share` or something like that if it doesn't return a boolean.